src/functTest/groovy/com/bmuschko/gradle/docker/DockerRemoteApiPluginFunctionalTest.groovy

```diff
@@ -74,6 +81,52 @@ class DockerRemoteApiPluginFunctionalTest extends AbstractGroovyDslFunctionalTes
         build('verify')
     }
 
+    @Requires({ TestPrecondition.DOCKER_PRIVATE_SECURE_REGISTRY_REACHABLE })
+    def "can push and pull images overriding wrong registryCredentials"() {
+        given:
+        buildFile << """
+            import com.bmuschko.gradle.docker.tasks.RegistryCredentialsAware
+            import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
+            import com.bmuschko.gradle.docker.tasks.image.DockerPushImage
+            import com.bmuschko.gradle.docker.tasks.image.DockerPullImage
+            import com.bmuschko.gradle.docker.tasks.image.Dockerfile
+
+            task dockerfile(type: Dockerfile) {
+                from '$TEST_IMAGE_WITH_TAG'
+            }
+
+            task buildImage(type: DockerBuildImage, dependsOn: dockerfile) {
```

Comment:
You can defining the task dependency twice. Can you remove it from here?

Reply:
done. I also changed the way other task dependencies are defined to avoid mixed definitions